### PR TITLE
Add a proposal for a TensorFlow backend.

### PR DESCRIPTION
**Pull request by hodgestar**

### CODE REVIEW

File: development/ideas/tensorflow-data-backend.rst [hunk 1]
```diff
@@ -23,6 +23,9 @@ efficiently. We hope that by using TensorFlow we might enable QuTiP to scale
 to bigger quantum systems (e.g. more qubits) and decrease the time taken to
 simulate them.
 
+There is particular interest in the trying the new backend with the
+BoFiN HEOM (Hierarchical Euqations of Motion) solver with the new backend [2]_.
```
Comment (nwlambert) [inline development/ideas/tensorflow-data-backend.rst:27]: Euqations = Equations..
Backend repeated twice...

Comment (hodgestar) [reply development/ideas/tensorflow-data-backend.rst]: Fixed! Thank you for catching those. Doh.

File: development/ideas/tensorflow-data-backend.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+Why a TensorFlow backend?
+-------------------------
+
+TensorFlow supports distributing matrix operations across multiple GPUs and
+multiple machines, and abstracts away some of the complexities of doing so
+efficiently. We hope that by using TensorFlow we might enable QuTiP to scale
+to bigger quantum systems (e.g. more qubits) and decrease the time taken to
+simulate them.
+
+There is particular interest in the trying the new backend with the
```
Comment (nathanshammah) [inline development/ideas/tensorflow-data-backend.rst:26]:
```suggestion
There is particular interest in trying the new backend with the
```

File: development/ideas/tensorflow-data-backend.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+differences, and it is possible that this will prove challenging or even
+that we will not find a reasonable way to achieve the desired performance.
+
+Expected outcomes
+=================
+
+* Add a `qutip.core.data.tensorflow` data type.
+* Implement specialisations for some important operations (e.g. `add`,
+  `mul`, `matmul`, `eigen`, etc).
+* Write a small benchmark to show how QObj operations scale on the new
```
Comment (nathanshammah) [inline development/ideas/tensorflow-data-backend.rst:47]:
```suggestion
* Write a small benchmark to show how Qobj operations scale on the new
```

**Review by nathanshammah — CHANGES_REQUESTED** [review]

**Comment by nathanshammah** [comment]
I had added some comments, I see this was merged already. @hodgestar

**Comment by jakelishman** [comment]
missed it by less than a minute apparently! Typos aside, this also looked fine to me.

**Comment by nwlambert** [comment]
Sorry, I was a bit too quick on the button!

**Comment by hodgestar** [comment]
@nathanshammah Thanks for reviewing! I made #134 to address your comments.

**Comment by nathanshammah** [comment]
I added this project idea to the GSoC 2021 wiki,
https://github.com/qutip/qutip/wiki/Google-Summer-of-Code-2021#tensorflow-data-backend
 using pandoc to convert from rst to md.

**Pull request merged**